src/main/java/seedu/ibook/model/product/DiscountRate.java

```diff
@@ -0,0 +1,87 @@
+package seedu.ibook.model.product;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.ibook.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Product's discount rate in the ibook.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDiscountRate(String)}
+ */
+public class DiscountRate {
+
+    public static final String DEFAULT_DISCOUNTRATE = "0.00";
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Discount Rate should only be a positive double, and at most 100";
+
+    public static final String VALIDATION_REGEX = "|(\\d+(?:.\\d{1,2})?)%?";
```

Comment:
Missing an escape sequence
```suggestion
    public static final String VALIDATION_REGEX = "|\\d+(?:\\.\\d{1,2})?%?";
```

Also, should we accept inputs like `.12`?

Reply:
Nice catch on that! 

> Also, should we accept inputs like .12?
I think might be okay to don't accept inputs in this form. 
Also, the use case for 0.xx% discount should also be quite rare.